client.go

```diff
@@ -432,6 +432,48 @@ func DefaultRetryPolicy(ctx context.Context, resp *http.Response, err error) (bo
 	return false, nil
 }
 
+// ErrorPropagatedRetryPolicy is the same as DefaultRetryPolicy, except it
+// propagates errors back instead of returning nil. This allows you to inspect
+// why it decided to retry or not.
+func ErrorPropagatedRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	// do not retry on context.Canceled or context.DeadlineExceeded
+	if ctx.Err() != nil {
```

Comment:
I think this should use the non-blocking check on ctx here instead, ie: ```
select {
case <-ctx.Done():
   return false, ctx.Err()
default:
  // ok, keep going
}
```

Reply:
@evanphx I can change it if you'd prefer it. The contract as specified in the `context.Context` interface indicates it's not necessary:

```go
// If Done is not yet closed, Err returns nil.
// If Done is closed, Err returns a non-nil error explaining why:
// Canceled if the context was canceled
// or DeadlineExceeded if the context's deadline passed.
// After Err returns a non-nil error, successive calls to Err return the same error.
```

My understanding is that the channel mechanism is for when you want to use a `select` between multiple channels, one being the context's `Done()` channel. 

Please let me know if you want the change before merging.

Reply:
Ah! I had forgotten the docs said that. No problem, leave as is.